module/zfs/dmu_object.c

```diff
@@ -59,10 +59,18 @@ dmu_object_alloc_dnsize(objset_t *os, dmu_object_type_t ot, int blocksize,
 	dnode_t *dn = NULL;
 	int dn_slots = dnodesize >> DNODE_SHIFT;
 	boolean_t restarted = B_FALSE;
-	uint64_t *cpuobj = &os->os_obj_next_percpu[CPU_SEQID %
-	    os->os_obj_next_percpu_len];
+	uint64_t *cpuobj = NULL;
 	int dnodes_per_chunk = 1 << dmu_object_alloc_chunk_shift;
 
+	/*
+	 * Pick the current CPU. We need to disable preemption to ensure that
+	 * we don't get interrupted (and pick the wrong CPU).
+	 */
```

Comment:
In this case it isn't really about picking the wrong cpu, it's about using the  current cpu as a hint to minimize lock contention.  The code is written is such a way that it's safe for this function to be rescheduled at any time to a different cpu.  I'd suggest dropping the comment here to avoid any confusion.